geoh5py/ui_json/validation.py

```diff
@@ -52,10 +52,8 @@ def __init__(
         ui_json: dict[str, Any] = None,
         **validation_options,
     ):
-        self._inferred_validations = (
-            {} if ui_json is None else self.infer_validations(ui_json)
-        )
-        self.validations: dict[str, Any] | None = validations
+        self.validations = {} if ui_json is None else self.infer_validations(ui_json)
+        self.validations = self._merge_validations(validations)
```

Comment:
Not a big fan on this syntax, sort of repeating the same assignment twice in a row. And this would brake if validations is None since not protected in _merge_validations

Reply:
Updated this section as we discussed: passing validations into infer_validations and handling internally